agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1652,7 +1654,8 @@ bool backhaul_manager::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_r
         // Handle the CMDU message. If the message was processed locally
         // (by the Backhaul Manager), this function will return 'true'.
         // Otherwise, it should be forwarded to the slaves.
-        if (handle_1905_1_message(cmdu_rx, src_mac)) {
+        Socket *destinationAP = nullptr;
+        if (handle_1905_1_message(cmdu_rx, src_mac, destinationAP)) {
```

Comment:
I agree, this forwarding mechanism is error prone, mostly because the return value true/false does not indicates success or fail in this case, but rather it signals the caller function what it should do next.
I think @morantr is working on rewriting this area.